tailwind/ui/buttons/Button.tsx

```diff
@@ -0,0 +1,149 @@
+import * as React from "react"
+import { cva, type VariantProps } from "class-variance-authority"
+import { Slot } from "@radix-ui/react-slot"
+
+import { BaseLink, type LinkProps } from "@/components/Link"
+
+import { cn } from "@/lib/utils/cn"
+import { type MatomoEventOptions, trackCustomEvent } from "@/lib/utils/matomo"
+import { scrollIntoView } from "@/lib/utils/scrollIntoView"
+
+const buttonVariants = cva(
+  "inline-flex gap-2 items-center justify-center rounded border-[1px] border-solid border-current text-primary transition focus-visible:outline focus-visible:outline-4 focus-visible:outline-primary-hover focus-visible:-outline-offset-1 disabled:text-disabled disabled:pointer-events-none hover:text-primary-hover [&[data-secondary='true']]:text-body",
+  {
+    variants: {
+      variant: {
+        solid:
+          "!text-background bg-primary border-transparent disabled:bg-disabled disabled:text-background hover:text-background hover:bg-primary-hover hover:shadow-button-hover active:shadow-none",
+        outline: "hover:shadow-button-hover active:shadow-none",
+        ghost: "border-transparent",
+        link: "border-transparent font-bold underline py-0 px-1 active:text-primary",
```

Comment:
This won't have any effect, I would just remove them to keep this cleaner.

IMO this should be controlled inside the `sizes`, I wouldn't add anything yet though but, in case we need it, I think it would be better to add a `link` or `inline` size.
```suggestion
        link: "border-transparent font-bold underline active:text-primary",
```

Reply:
@pettinarip Can do a [compound variant](https://cva.style/docs/getting-started/variants#compound-variants) here

Reply:
Makes sense! what do you have in mind? something like this?

```ts
// could be weird because we are not composing nothing but could work really well
{
  variant: "link",
  className: "py-0 px-1"
}
```

Reply:
@pettinarip Something like that.

Better than 
```tsx
// potential redundancy 
<Button variant="link" size="link" />
```

Also, when are we expected to use this variant when there is a `ButtonLink` component and we just use a normal link anyway inline.

Reply:
Got it, agree. Ok, I think once we add that we should be good to go. Nice job!

Reply:
@pettinarip sorry, that last bit was supposed to be a question! 😅

Reply:
Ahh got it xD

Good point, this is a really minor use case that we don't use. I just marked it to be consistent with the previous implementation.

We can skip it for now and add it later if we find it more relevant.